File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int low = 0,high = 0;
        int n = nums.length ;
        int res = Integer.MIN_VALUE ;
        int zeroes = 0;

        while(high<n){
            if(nums[high] == 0){
                zeroes++;
            }

            while(zeroes>k){
                if(nums[low] == 0){
                    zeroes--;

                }
                    low++;
            }
            int len = high-low+1;
            res = Math.max(len,res);
            high++;


        }
        return res ;

       
        
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int left=0,right=0,maxi = INT_MIN ,zeroes = 0;

        for(right=0;right<nums.size();right++){
            if(nums[right] == 0){
                zeroes++;
            }

            while(zeroes> k){
                if(nums[left] ==0){

                zeroes--;
                }
                left++ ;

            }
            int len = right-left+1 ;
            maxi = max(len,maxi) ;

        }

        return maxi ;
        
    }
};